src/sections/Home/components/AnimatedRole/HomeRole.js

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { makeStyles, Typography } from "@material-ui/core";
+import { styles } from "../../../../styles/styles";
+
+const useStyles = makeStyles(theme => ({
+  text2: {
+    color: "white",
+    fontFamily: "Roboto Condensed, sans-serif",
+    [theme.breakpoints.only("xs")]: {
+      fontSize: "1.1rem"
+    },
+    [theme.breakpoints.only("sm")]: {
+      fontSize: "1.7rem"
+    },
+    [theme.breakpoints.only("md")]: {
+      fontSize: "2rem"
+    },
+    [theme.breakpoints.only("lg")]: {
+      fontSize: "2.125rem"
+    },
+    [theme.breakpoints.only("xl")]: {
+      fontSize: "3.4rem"
+    }
+  },
+  cursor1: {
+    borderLeft: `${styles.sizes.thickness.thick2} solid gray`
+  },
+  cursor2: {
+    borderLeft: `${styles.sizes.thickness.thick2} solid transparent`
+  },
+  cursorHidden: {
+    visibility: "hidden"
+  }
+}));
+
+const AnimatedRole = ({ messages }) => {
+  const classes = useStyles();
+  const [state, setState] = useState({
```

Comment:
Sepparate the state into different states asi this too generic and any change would cause the component to re-render. Only group variables into the same state if they are part of a group that makes sense